modules/gridmap/grid_map.cpp

```diff
@@ -115,7 +120,10 @@ bool GridMap::_get(const StringName &p_name, Variant &r_ret) const {
 		Array ret;
 		ret.resize(baked_meshes.size());
 		for (int i = 0; i < baked_meshes.size(); i++) {
-			ret[i] = baked_meshes[i].mesh;
+			Array pair;
+			pair.append(baked_meshes[i].mesh);
+			pair.append(baked_meshes[i].item);
+			ret[i] = pair;
```

Comment:
The problem pointed out by Malcom comes from doing this. You are changing the interpretation of the "baked_meshes" property in the .tscn file.

If you add more data, you need to ensure compatibility. You can do that in two ways:
1. Add the data without changing how old properties are interpreted (i.e. add a new property)
2. In the `_set()` function check the data type that you read, and then gracefully handle upgrading old scenes. I.e. check if `meshes[0]` is an Array or a Mesh. If its a mesh, use the old codepath, if its an array then do the new codepath.

Reply:
Yes and I was expecting that old scene using baked meshes won't work after my change.

I am working now on version which use approach 1. (new property instead of updating old one), but I still need to do some testing and find the best way to fix that.